scripts/travis-install

```diff
@@ -1,34 +1,55 @@
 #!/bin/bash
-set -e
-set -x
+set -e  # Exit on non-zero status
+set -x  # Print commands run
 export LC_CTYPE=en_US.UTF-8
 
-if [ "$TOXENV" == "py27" ]
+# Download ElasticSearch
+if [ ${INSTALL_ELASTICSEARCH:-0} -ne 0 ]
 then
-    sudo apt-get update -qq
-    sudo apt-get -y install build-essential libxml2-dev libxslt-dev libjpeg8 libjpeg8-dev libfreetype6 libfreetype6-dev zlib1g-dev sqlite3 tidy libtidy-dev libtidy-0.99-0 python-dev libffi-dev libssl-dev
-    sudo ln -s /usr/lib/`uname -i`-linux-gnu/libfreetype.so /usr/lib
-    sudo ln -s /usr/lib/`uname -i`-linux-gnu/libz.so /usr/lib
+    mkdir -p downloads
+    wget -q -O downloads/elasticsearch-$ES_VERSION.tar.gz $ES_DOWNLOAD_URL
+    tar -zxf downloads/elasticsearch-$ES_VERSION.tar.gz
+fi
+
+# Run ElasticSearch in background
+(if [ ${INSTALL_ELASTICSEARCH:-0} -ne 0 ]
+then
+    ./elasticsearch-$ES_VERSION/bin/elasticsearch 1> elasticsearch.log 2> elasticsearch.err
+fi) &
 
-    # completely and utterly remove Travis' MySQL and let the Ansible role install it
-    sudo apt-get remove --purge 'mysql*'
-    sudo apt-get autoremove
-    sudo apt-get autoclean
-    sudo rm -rf /var/lib/mysql
 
-    # Install Elasticsearch
-    wget https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.7.2.deb
-    sudo dpkg -i elasticsearch-1.7.2.deb
+# Get database ready
+if [ -n "${CREATE_DB:-}" ]
+then
+    mysql -e "CREATE DATABASE IF NOT EXISTS ${CREATE_DB};";
+fi
 
-    # limit elasticsearch / java memory usage to avoid OOM Killer
-    sudo service elasticsearch stop;
-    echo "ES_HEAP_SIZE=256m" | sudo tee --append /etc/default/elasticsearch
-    sudo service elasticsearch start;
+# Install pipeline toolchain
+if [ ${INSTALL_PIPELINE:-0} -ne 0 ]
+then
+    npm install
+    npm install cssmin@0.4.3
+    npm install uglify-js@2.4.13
 fi
 
-if [ "$TOXENV" == "docker" -o "$TOXENV" == "locales" ]
+# Install docker-compose
+if [ ${INSTALL_DOCKER_COMPOSE:--} -ne 0 ]
 then
-    curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
-    chmod +x docker-compose
-    sudo mv docker-compose /usr/local/bin
+    if [ -x $(command -v docker-compose) ]
+    then
+        echo "Overwriting existing docker-compose."
+        docker-compose -v
+    else
+        echo "Installing docker-compose ${DOCKER_COMPOSE_VERSION}."
+    fi
+    mkdir -p downloads
+    DOCKER_COMPOSE_FILE=downloads/docker-compose-${DOCKER_COMPOSE_VERSION}
+    if [ ! -f $DOCKER_COMPOSE_FILE ]
+    then
+        wget -q --waitretry=1 --retry-connrefused -T 10 \
+            -O downloads/docker-compose-${DOCKER_COMPOSE_VERSION} \
+            https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m`
+    fi
+    chmod +x $DOCKER_COMPOSE_FILE
+    sudo cp $DOCKER_COMPOSE_FILE /usr/local/bin/docker-compose
```

Comment:
You can cache directories but not files. I think it would be a mistake to cache ``/usr/local/bin``, so I think we're OK caching the file in ``downloads`` and copying it each time.